keepalived-vip/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	flag "github.com/spf13/pflag"
+
+	"github.com/openshift/origin/pkg/util/proc"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	kubectl_util "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+var (
+	flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	cluster = flags.Bool("use-kubernetes-cluster-service", true, `If true, use the built in kubernetes
+        cluster for creating the client`)
+
+	logLevel = flags.Int("v", 1, `verbose output`)
+
+	useUnicast = flags.Bool("use-unicast", false, `use unicast instead of multicast for communication
+		with other keepalived instances`)
+
+	// sysctl changes required by keepalived
+	sysctlAdjustments = map[string]int{
+		"net/ipv4/ip_nonlocal_bind": 1,
+	}
+)
+
+func main() {
+	clientConfig := kubectl_util.DefaultClientConfig(flags)
+	flags.Parse(os.Args)
+
+	var err error
+	var kubeClient *unversioned.Client
+
+	if *cluster {
+		if kubeClient, err = unversioned.NewInCluster(); err != nil {
+			glog.Fatalf("Failed to create client: %v", err)
+		}
+	} else {
+		config, err := clientConfig.ClientConfig()
+		if err != nil {
+			glog.Fatalf("error connecting to the client: %v", err)
+		}
+		kubeClient, err = unversioned.New(config)
+	}
+
+	namespace, specified, err := clientConfig.Namespace()
+	if err != nil {
+		glog.Fatalf("unexpected error: %v", err)
+	}
+
+	if !specified {
+		namespace = ""
+	}
+
+	err = loadIPVModule()
+	if err != nil {
+		glog.Fatalf("Terminating execution: %v", err)
+	}
+
+	err = changeSysctl()
```

Comment:
Please add a comment about this, or a link to some explanation. Either here or above the function def is fine (or above sysctlAdjustments).